inst/htmljs/vendor/quadprog.js

```diff
@@ -598,75 +601,213 @@ function solveQP(Dmat, dvec, Amat, bvec = [], meq = 0, factorized = [0, 0]) {
     };
 }
 
-function optimizeAlignedBoxes(data, alignment, min_distance) {
-    console.log("got data : ", data, "alignement : ", alignment)
-  var n = data.length;
+// ---------------------------------------------------------------------------------------------------
+// Label positioning functions inspired by the R `directlabels` package: https://github.com/tdhock/directlabels
+//
+// `optimizeAlignedBoxes` uses a QP solver to reposition only the overlapping label boxes,
+// finding the nearest possible positions to their original locations, subject to the constraint
+// that the boxes do not overlap.
+//
+// `bumpup` and `bumpleft` are heuristic fallback strategies, also inspired by directlabels.
```

Comment:
i would prefer to not add bumpup and bumpleft. can you please remove? or justify their addition?